contracts/minter-suite/Minters/MinterSEAV0.sol

```diff
@@ -0,0 +1,1186 @@
+            uint24 nextTokenNumber,
+            Auction memory auction
+        )
+    {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        maxInvocations = _projectConfig.maxInvocations;
+        timestampStart = _projectConfig.timestampStart;
+        auctionDurationSeconds = _projectConfig.auctionDurationSeconds;
+        basePrice = _projectConfig.basePrice;
+        nextTokenNumberIsPopulated = _projectConfig.nextTokenNumberIsPopulated;
+        nextTokenNumber = _projectConfig.nextTokenNumberIsPopulated
+            ? _projectConfig.nextTokenNumber
+            : 0;
+        auction = _projectConfig.activeAuction;
+    }
+
+    /**
+     * @notice projectId => active auction details.
+     * @dev reverts if no auction exists for the project.
+     */
+    function projectActiveAuctionDetails(
+        uint256 _projectId
+    ) external view returns (Auction memory auction) {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        auction = _projectConfig.activeAuction;
+        // do not return uninitialized auctions (i.e. auctions that do not
+        // exist, and therefore are simply the default struct)
+        require(auction.initialized, "No auction exists on project");
+        return auction;
+    }
+
+    /**
+     * @notice Convenience function that returns either the current token ID
+     * being auctioned, or the next expected token ID to be auction if no
+     * auction is currently initialized or if the current auction has concluded
+     * (block.timestamp > auction.endTime).
+     * This is intended to be useful for frontends or scripts that intend to
+     * call `createBid` or `settleAuctionAndCreateBid`, which requires a
+     * target bid token ID to be passed in as an argument.
+     * The function reverts if a project does not have an active auction and
+     * the next expected token ID has not been populated.
+     * @param _projectId The project ID being queried
+     * @return The current token ID being auctioned, or the next token ID to be
+     * auctioned if a new auction is ready to be created.
+     */
+    function getTokenToBid(uint256 _projectId) external view returns (uint256) {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        Auction storage _auction = _projectConfig.activeAuction;
+        // if project has an active token auction that is not settled, return
+        // that token ID
+        if (_auction.initialized && (_auction.endTime > block.timestamp)) {
+            return _auction.tokenId;
+        }
+        // otherwise, return the next expected token ID to be auctioned
+        if (!_projectConfig.nextTokenNumberIsPopulated) {
+            revert("Next token number not populated");
+        }
+        // @dev overflow automatically checked in Solidity ^0.8.0
+        uint256 nextTokenId = (_projectId * ONE_MILLION) +
+            _projectConfig.nextTokenNumber;
+        return nextTokenId;
```

Comment:
Nit: For code-readability purposes, I'm tempted to just return value of `getNextTokenId` here instead.

Reply:
Love this, updated to do this in commit [4efdf77](https://github.com/ArtBlocks/artblocks-contracts/pull/504/commits/4efdf773a1d4054e5cdb8512cc8a6c4c81e3b71d)